.github/workflows/pr_cluster_workflow.yml

```diff
@@ -0,0 +1,174 @@
+      with:
+        workspace: ${{ env.TF_VAR_cluster_name }}-cluster
+        plan_output: tfplan-${{ env.TF_VAR_cluster_name }}-cluster
+        working_directory: "terraform/cluster"
+        tf_api_token: ${{ secrets.TF_API_TOKEN }}
+
+    - name: Generating kube-config
+      uses: actions/upload-artifact@v4
+      with:
+        name: kube-config-${{ env.TF_VAR_cluster_name }}
+        path: /opt/kubeconfig/${{ env.TF_VAR_cluster_name }}
+        compression-level: 0
+
+    - name: Generating Markdown
+      run: |
+        echo "### turnk8s" >> $GITHUB_STEP_SUMMARY
+        echo "" >> $GITHUB_STEP_SUMMARY
+        echo "Push your Kubernetes service manifests to [GitHub URL](https://github.com/infraheads/${{ env.TF_VAR_cluster_name }}) to get them deployed on the cluster. :star_struck:" >> $GITHUB_STEP_SUMMARY
+        echo "Use `kubeconfig` file(s) to connect to the cluster, see the link above :point_up:" >> $GITHUB_STEP_SUMMARY
+
+  rollback_cluster:
+    needs: update_cluster
+    if: ${{ needs.update_cluster.result == 'failure' }}
+    runs-on: self-hosted
+    permissions:
+      contents: read
+      pull-requests: write
+    defaults:
+      run:
+        working-directory: "terraform/infrastructure"
+    steps:
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.14.0
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v3
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_version: 1.7.5
+
+      - name: Configure Terraform Cache
+        run: echo "TF_PLUGIN_CACHE_DIR=$HOME/.terraform.d/plugin-cache" >> "$GITHUB_ENV"
+
+      - name: Initializing Terraform
+        run: terraform init
+        env:
+          TF_WORKSPACE: "default-ws"
+
+      - name: Selecting Terraform Workspace
+        run: terraform workspace select ${{ env.TF_VAR_cluster_name }}-infrastructure
+
+      - name: Destroying Terraform
+        run: |
+          terraform destroy -auto-approve
+
+      - name: Destroying Terraform Cloud Workspace
+        run: |
+          terraform workspace select default-ws
+          terraform workspace delete -force ${{ env.TF_VAR_cluster_name }}-infrastructure
+          terraform workspace delete -force ${{ env.TF_VAR_cluster_name }}-cluster
+
+      - name: Remove kubeconfig
+        run: rm /opt/kubeconfig/${{ env.TF_VAR_cluster_name }}
+
+  ready_to_merge:
+    needs: [setup_cluster, update_cluster]
+    if: ${{ needs.setup_cluster.result != 'failure' && needs.update_cluster.result != 'failure' }}
+    runs-on: self-hosted
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+    - name: Block PR merge if no cluster names
+      run: |
+        if [[ ${{ needs.setup_cluster.outputs.clusters }} == '[]' ]]; then
+          if [[ ${{ github.event.pull_request.draft }} == true ]]; then
+            echo "There is no cluster with name ${{ env.TF_VAR_cluster_name }}."
```

Comment:
You try to handle a type-in in the config.yaml for the test cluster name. The message should clearly state that and nothing else.